README.md

```diff
@@ -3,44 +3,164 @@
+            authorization-grant-type: authorization_code
+            redirect-uri: "{baseUrl}/login/oauth2/code/{registrationId}"
+            scope: openid, profile
+            provider: idsvr
+
+client:
+  authentication:
+    asymmetric-key:
+      key-store-file-name: demo-private-jwt-client.jks
+      key-store-password: changeit
+      key-store-alias: demo-client
+      key-store-type: jks
+```
+
+Don't forget to point to the Curity Identity Server as well. Specify the issuer URI (by default, this is the anonymous endpoint of the Curity Identity Server). Spring Boot uses this URI to load the OpenID Connect metadata from the server.
+
+```yaml
+spring:
+  security:
+    oauth2:
+      client:
         provider:
           idsvr:
-            issuer-uri: https://idsvr.example.com/oauth/anonymous
+            issuer-uri: https://idsvr.example.com/oauth/v2/oauth-anonymous
 ```
 
+See [application.yml](src/main/resources/application.yml) for the complete configuration.
+
 ## Run the application
 To start the application run 
 
 ```bash
 ./gradlew bootRun
 ```
 
-Open `http://localhost:8080` in your browser. Click on the link to login and fetch an access and ID token from the Curity Identity Server.
+Open `http://localhost:8080` in your browser. Click on the link to log in. 
+Open one of the different options to fetch an access and ID token from the Curity Identity Server. 
+After successful login the page displays the username and the name of the client used to integrate with the Curity Identity Server.
+
+### Configuring the Trust Store
+
+The application must trust the HTTPS server certificate of the Curity Identity Server. Place the issuing CA certificate of the server certificate and all intermediate CA certificate as well as the root CA certificate in a trust store. 
+
+Start the application with the truststore using JVM arguments:
+
+```bash
+./gradlew bootRun -Djavax.net.ssl.trustStore=/path/to/localhost.truststore -Djavax.net.ssl.trustStorePassword=changeit
+```
+
+## Generate Key and Trust Stores
+
+### Key Store for API Client Authentication
+
+When authenticating with `private_key_jwt`, you need to specify keys to issue the self-signed JWT for the authentication. Use `keytool` to generate a key pair:
+
+```bash
+keytool -genkey -alias demo-client -keyalg RSA -keystore demo-private-jwt-client.jks -keysize 2048 -dname "CN=Demo Client,O=Example"
+```
+
+The subject name of the certificate does not matter in this context. Make sure to remember the alias as it is used to identify the key pair. Put the key store file in the `resources` folder and update the parameters in `application.yml` accordingly.
+
+Export the certificate with the public key:
+
+```bash
+keytool -exportcert -keystore demo-private-jwt-client.jks -alias demo-client -file demo-client.cer
+```
+
+Upload the certificate file to the Curity Identity Server as a **Signature Verification Key** and configure the client with that key.
```

Comment:
Maybe add `in the **Facilities** -> **Keys and Cryptography** -> **Signing** section`